### PR TITLE
Term signal

### PR DESCRIPTION
**Pull request by benbjohnson**
## Overview

This pull request changes raft so that term changes are made immediately and term change signals are made afterward. Previously, election timeouts were invalidated by incoming term changes which caused an election loop.

Stale term was also fixed and http/pprof was added too.

### CODE REVIEW

File: cmd/influxd/handler.go [hunk 1]
```diff
@@ -29,6 +30,21 @@ func NewHandler() *Handler {
...
+		case "/debug/pprof/profile":
+			pprof.Profile(w, r)
+		case "/debug/pprof/symbol":
+			pprof.Symbol(w, r)
+		default:
+			pprof.Index(w, r)
+		}
+		return
+	}
+
```
Comment (benbjohnson) [inline cmd/influxd/handler.go]: You can now pull pprof profiles if you run:

```sh
$ go tool pprof http://localhost:8086/debug/pprof/profile
```

More info on the [`net/http/pprof`](https://golang.org/pkg/net/http/pprof/) page. Make sure you change their `:6060` ports to `:8086` though.

File: raft/log.go [hunk 2]
```diff
@@ -538,10 +538,20 @@ func (l *Log) setTerm(term uint64) error {
...
 	if err := l.setTerm(term); err != nil {
 		panic("unable to set term: " + err.Error())
 	}
+
+	// Signal term change.
+	select {
+	case l.terms <- struct{}{}:
+	default:
+	}
 }
```
Comment (benbjohnson) [inline raft/log.go]: Term updates immediately but signals asynchronously.

File: server.go [hunk 3]
```diff
@@ -1771,6 +1771,7 @@ func (s *Server) WriteSeries(database, retentionPolicy string, points []Point) (
 		for _, p := range points {
 			measurement, series := db.MeasurementAndSeries(p.Name, p.Tags)
 			if series == nil {
+				s.Logger.Printf("series not found: name=%s, tags=%#v", p.Name, p.Tags)
 				return ErrSeriesNotFound
 			}
```
Comment (benbjohnson) [inline server.go]: Additional logging when a series is not found.

**Comment by otoolep** [comment]
+1

**Pull request merged**